src/test/java/org/openrewrite/java/migrate/javaee/JavaEE7OpenJPATests.java

```diff
@@ -0,0 +1,231 @@
+/*
+ * Copyright 2024 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.openrewrite.java.migrate.javaee;
+
+import org.junit.jupiter.api.Test;
+import org.openrewrite.DocumentExample;
+import org.openrewrite.InMemoryExecutionContext;
+import org.openrewrite.config.Environment;
+import org.openrewrite.java.JavaParser;
+import org.openrewrite.test.RecipeSpec;
+import org.openrewrite.test.RewriteTest;
+
+import static org.openrewrite.java.Assertions.java;
+
+class JavaEE7OpenJPATests implements RewriteTest {
+    @Override
+    public void defaults(RecipeSpec spec) {
+        spec.parser(JavaParser.fromJavaVersion()
+            .classpathFromResources(new InMemoryExecutionContext(), "javax.persistence-api-2.2")).
+          recipe(Environment.builder().scanRuntimeClasspath("org.openrewrite.java.migrate.javaee7")
+            .build()
+            .activateRecipes("org.openrewrite.java.migrate.javaee7.openJPA"));
```

Comment:
Instead of parsing Yaml and having a level of indirection between the test class name and what Java recipe is tested, I'd prefer instead to have a dedicated test class for the Java recipe instead, which is directly created here through a constructor.

Having a test class name follow the tested class allows one to jump from the implementation to the tests through `Ctrl + Shift + T` in Intellij. Right now when I do that I get a popup that suggests there's no such test. 
![image](https://github.com/openrewrite/rewrite-migrate-java/assets/1027334/05d824ce-0a1f-4da0-8830-c612390c6b70)